terraform/modules/cloud-build-docker/build_image.py

```diff
@@ -54,6 +54,42 @@ def get_image_digest(image_uri, tag, project_id):
         return None
 
 
+def check_cache_tag_exists(image_uri, cache_tag, project_id):
+    """Check if a cache tag exists for the given image."""
+    try:
+        result = run_command(
+            [
+                "gcloud",
+                "container",
+                "images",
+                "list-tags",
+                image_uri,
+                "--filter",
+                f"tags:{cache_tag}",
+                "--limit",
+                "1",
+                "--format=get(digest)",
+                "--project",
+                project_id,
+            ]
+        )
+        return bool(result.stdout.strip())
+    except subprocess.CalledProcessError:
+        return False
+
+
+def get_effective_cache_tag(image_uri, image_tag_suffix, project_id):
+    """Determine the effective cache tag, falling back to 'latest' if the provided tag doesn't exist."""
```

Comment:
```suggestion
    """Determine the effective cache tag.
    
    Ideally, the cache tag we use is the requested tag, `image_tag_suffix`.
    But if that doesn't exist, we fall back to 'latest'.  In that case, we presumably
    won't be able to use the full image as a cache (it doesn't have the right tag)
    but we will hopefully be able to use some of its layers, saving *some* time.
    """
```

(I'm just guessing that this is what is going on.  If not, please modify the comments appropriately.)